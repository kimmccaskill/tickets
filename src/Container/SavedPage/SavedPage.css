.saved-page-title {
  margin-top: 2em;
  margin-left: 2%;
  padding-left: 1em;
  border-left: #CAD5CA 5px solid;
}

.error {
  margin: 2%;
}