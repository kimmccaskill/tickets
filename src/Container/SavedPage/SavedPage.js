import React from 'react'
import './SavedPage.css'
import EventsContainer from '../../Components/EventsContainer/EventsContainer.js'
import { connect } from 'react-redux';

export const SavedPage = ({ savedEvents }) => {
  const checkSaved = () => {
    return savedEvents.length > 0 ? <EventsContainer events={savedEvents}/> : <h2 className='error'>There are no saved events.</h2>
  }

  return (
    <section className='saved-page'>
      <h2 className='saved-page-title'>Your Saved Events</h2>
      {checkSaved()}
    </section>
  )
}

export const mapStateToProps = state => ({
  savedEvents: state.savedEvents
})

export default connect(mapStateToProps)(SavedPage);