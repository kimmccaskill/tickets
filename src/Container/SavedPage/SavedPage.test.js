import React from 'react';
import { SavedPage, mapStateToProps } from './SavedPage';
import { shallow } from 'enzyme';
import { mockEvents } from '../../mockData'

describe('SavedPage', () => {
  let wrapper,mockState
  it('should match the snapshot', () => {
    wrapper = shallow(<SavedPage savedEvents={mockEvents}/>);

    expect(wrapper).toMatchSnapshot();
  });
  
  describe('mapStateToProps', () => {
    it('should return an object with savedEvents', () => {
      mockState = {savedEvents: mockEvents}

      const expected = {savedEvents: mockEvents}
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected)
    })
  })
});