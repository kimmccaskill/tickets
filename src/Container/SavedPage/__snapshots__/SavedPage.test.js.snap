// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedPage should match the snapshot 1`] = `ShallowWrapper {}`;
